rs/nervous_system/integration_tests/tests/sns_upgrade_test_utils_legacy.rs

```diff
@@ -42,6 +44,31 @@ pub async fn test_sns_upgrade_legacy(sns_canisters_to_upgrade: Vec<SnsCanisterTy
     )
     .await;
 
+    eprintln!("Await the swap lifecycle ...");
+    sns::swap::await_swap_lifecycle(&pocket_ic, sns.swap.canister_id, Lifecycle::Open)
+        .await
+        .unwrap();
+
+    eprintln!("smoke_test_participate_and_finalize ...");
+    sns::swap::smoke_test_participate_and_finalize(
+        &pocket_ic,
+        sns.swap.canister_id,
+        swap_parameters,
+    )
+    .await;
+
+    eprintln!(
+        "Disabling automatic upgrades to have full control over when an upgrade is triggered ..."
+    );
+    let automatically_advance_target_version = false;
```

Comment:
Note to reviewer. since this is a legacy test (it tests upgradability via `UpgradeSnsToNextVersion` proposals), we only maintain a version that's equivalent to the previous one.